### DISCUSSION TITLE
how to intercept submitted form data

### DISCUSSION THREAD

**Discussion by asimakos** [MasaCMS/MasaCMS#256] [General]
Hello!

I would like to ask you if there is any way to intercept submitted form data according to Steve Withington's mura contacts example. Unfortunately `m.event('formfield')` does not seem to work at all for form scope at least for me, only for url scope works fine.

Regards

**Comment by Tigerlady** [comment]
What exactly are you trying to do? I have what I call a pretty email that gets sent to the client instead of the standard one from Masa. In the event Handler file, I have something like this. 
`<cffunction name="onAfterFormSubmitSave" output="false">
<cfargument name="$">
<cfset receipt = $.event('formDataBean').getFormResult() />
	

<cfset record = receipt />`


Then all I have to do is something like this to capture the form data
`#form.yourname#<br />
#form.company#<br />
#form.Email#<br />
</cffunction>  
</cfcomponent>`

Of course, I add styling and other things, but using the onAfterFormSubmitSave allows me to capture the submitted data. 

I hope this helps.

**Comment by Tigerlady** [comment]
For some reason, the editor took out the cffunction lines. 
![image](https://github.com/MasaCMS/MasaCMS/assets/1850313/086e25a1-5a5a-4f00-b2c9-f4cbe086c127)

**Reply by asimakos** [reply]
Hello!
I try to play around with a custom form CRUD app similar to Steve withington's mura contacts (cfml version) https://github.com/stevewithington/muracontacts/tree/cfml
and unfortunately I have met this type of error only for form scope. Could you share your code repository as it would help me a lot especially with my crud actions?

Regards again

**Comment by asimakos** [comment]
Hello!

I played with your code but unfortunately for me there is no `getFormResult` method and additionally `$.event('formDataBean')` returns null via cfdump. I just added some other similar functions from [here](https://gist.github.com/stevewithington/8314e97d136ad23769de) and still the same issue.  

I think that it might be a problem regarding my installation or possibly a setting that i do not know. For better convenience i attach code files 

https://we.tl/t-4aa04AZVnr

as well as masa cms (box.json).

[box.json](https://github.com/MasaCMS/MasaCMS/files/13376729/box.json)

Regards

**Comment by asimakos** [comment]
My app works fine just with `m.event('formfield')`. I only had to remove and add it again via inline edit. I do not know what happened previously ?

Regards again
